daemon/algod/api/server/v2/handlers.go

```diff
@@ -1393,12 +1393,22 @@ func (v2 *Handlers) getPendingTransactions(ctx echo.Context, max *uint64, format
 }
 
 // startCatchup Given a catchpoint, it starts catching up to this catchpoint
-func (v2 *Handlers) startCatchup(ctx echo.Context, catchpoint string) error {
-	_, _, err := ledgercore.ParseCatchpointLabel(catchpoint)
+func (v2 *Handlers) startCatchup(ctx echo.Context, catchpoint string, initializeRounds uint64) error {
+	catchpointRound, _, err := ledgercore.ParseCatchpointLabel(catchpoint)
 	if err != nil {
 		return badRequest(ctx, err, errFailedToParseCatchpoint, v2.Log)
 	}
 
+	if initializeRounds > 0 {
+		ledgerRound := v2.Node.LedgerForAPI().Latest()
```

Comment:
I think need a test for this condition to have this behavior "fixed".

Reply:
There's a new unit test in this PR. How would you like it to be improved?

Reply:
I mismatched `"the node has already been initialized"` check and the  error`errCatchpointWouldNotInitialize`. Looks all right.